src/libraries/System.IO.FileSystem/tests/PortedCommon/IOInputs.cs

```diff
@@ -5,21 +5,20 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using System.Runtime.InteropServices;
 
 internal static class IOInputs
 {
-    public static bool SupportsSettingCreationTime { get { return RuntimeInformation.IsOSPlatform(OSPlatform.Windows); } }
-    public static bool SupportsGettingCreationTime { get { return RuntimeInformation.IsOSPlatform(OSPlatform.Windows) | RuntimeInformation.IsOSPlatform(OSPlatform.OSX); } }
+    public static bool SupportsSettingCreationTime => OperatingSystem.IsWindows();
+    public static bool SupportsGettingCreationTime => OperatingSystem.IsWindows() | OperatingSystem.IsMacOS();
```

Comment:
Is there a reason the existing code uses `|` instead of `||`?

Reply:
I walked the blame and it's been like that all along, [going back to 2015](https://github.com/dotnet/corefx/pull/2433/files#diff-8cf3ac4125f3237e2b0d429133465401R18). The code's been refactored and updated several times along the way and this has persisted.

I've never really inspected which would be more efficient, but it looks like the bitwise OR is slightly more efficient.

```c#
public class C {
    public static bool BitwiseSupportsGettingCreationTime => OperatingSystem.IsWindows() | OperatingSystem.IsMacOS();
    public static bool LogicalSupportsGettingCreationTime => OperatingSystem.IsWindows() || OperatingSystem.IsMacOS();
}
```
``` asm
C.get_BitwiseSupportsGettingCreationTime()
    L0000: push ebp
    L0001: mov ebp, esp
    L0003: sub esp, 8
    L0006: cmp dword ptr [0x155cc1a8], 0
    L000d: je short L0014
    L000f: call 0x615f0140
    L0014: call OperatingSystem.IsWindows()
    L0019: mov [ebp-4], eax
    L001c: call OperatingSystem.IsMacOS()
    L0021: mov [ebp-8], eax
    L0024: mov eax, [ebp-4]
    L0027: or eax, [ebp-8]
    L002a: movzx eax, al
    L002d: mov esp, ebp
    L002f: pop ebp
    L0030: ret

C.get_LogicalSupportsGettingCreationTime()
    L0000: push ebp
    L0001: mov ebp, esp
    L0003: sub esp, 0xc
    L0006: cmp dword ptr [0x155cc1a8], 0
    L000d: je short L0014
    L000f: call 0x615f0140
    L0014: call OperatingSystem.IsWindows()
    L0019: mov [ebp-4], eax
    L001c: cmp dword ptr [ebp-4], 0
    L0020: jne short L0032
    L0022: call OperatingSystem.IsMacOS()
    L0027: mov [ebp-0xc], eax
    L002a: mov eax, [ebp-0xc]
    L002d: mov [ebp-8], eax
    L0030: jmp short L0039
    L0032: mov dword ptr [ebp-8], 1
    L0039: mov eax, [ebp-8]
    L003c: movzx eax, al
    L003f: mov esp, ebp
    L0041: pop ebp
    L0042: ret
```

Reply:
Maybe worth opening a codegen issue.

Reply:
I just looked at the [docs](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#logical-or-operator-) to be 100% sure and the difference for `|` and `||` for `bool` is very small:

> The | operator evaluates both operands even if the left-hand operand evaluates to true, so that the operation result is true regardless of the value of the right-hand operand.

It looks like it does not make any sense to use `|` over `||` here. I am going to change it

Reply:
> The | operator evaluates both operands even if the left-hand operand evaluates to true

That part is usually my concern when seeing code that uses `bool | bool`. Sometimes the right-hand operand does unnecessary work. Or it might even be a bug like: `if (myVar == null | myVar.SomeProperty)`. IMO the default should be `||` unless there is a reason to use `|` (like if you are OR'ing bits, or you really want both sides to be evaluated).